### ISSUE TITLE
How to reference Ladybug + Honeybee in the publications

### ISSUE THREAD

**Issue by mostaphaRoudsari** [mostaphaRoudsari/ladybug#133]
It is currently not clear at all how people should reference to the development in their publications. I have seen some really surprising versions recently-

@chriswmackey here is my initial version for your comments. For now I use the paper that I published last year for Ladybug until we write the new paper for the coming IBPSA conference.

Ladybug and Honeybee are two open source environmental plugins for Grasshopper3D for Rhino. Ladybug imports standard EnergyPlus Weather files (.EPW) into Grasshopper3D and provides a variety of 3D interactive graphics to support the decision-making process during the initial stages of design. Honeybee connects Grasshopper3D to EnergyPlus, Radiance, Daysim and OpenStudio for building energy and daylighting simulation. The Honeybee project intends to make many of the features of these simulation tools available in a parametric way (Sadeghipour Roudsari M., Pak M., 2013).

Roudsari, Mostapha Sadeghipour, Pak, Michelle, 2013. Ladybug: a parametric environmental plugin for grasshopper to help designers create an environmentally-conscious design. In: Proceedings of the 13th International IBPSA Conference Held in Lyon, France Aug 25–30th. (http://www.ibpsa.org/proceedings/BS2013/p_2499.pdf)

**Comment by chriswmackey** [comment]
@mostaphaRoudsari ,

This sounds good.  The only suggestion that I would have is to change some things in the last sentence.  First, it might be a bit clearer to replace "in a parametric way" with "parametrically".  Also, from our conversations, I understand what you mean by parametric but I think that we should be aware that the word is used to describe a lot of different things these days and I was unclear on what you meant by it until you explained it.  To draw an analogy, I sometimes feel that the word "parametric" is on the same trajectory as "sustainable" and is suffering the same loss of meaning:
http://xkcd.com/1007/

For the sake of clarity, it might be good to add to the end of the last sentence:
"The Honeybee project aims to make the features of these simulation tools available parametrically and with the flexibility offered by a component-based, visual scripting interface."

Feel free to change it and let me know your thoughts,
-Chris

**Comment by mostaphaRoudsari** [comment]
Thanks @chriswmackey - You know that I love that cartoon! :) So here is the modified version based on your inputs:

Ladybug and Honeybee are two open source environmental plugins for Grasshopper3D for Rhino. Ladybug imports standard EnergyPlus Weather files (.EPW) into Grasshopper3D and provides a variety of 3D interactive graphics to support the decision-making process during the initial stages of design. Honeybee connects Grasshopper3D to EnergyPlus, Radiance, Daysim and OpenStudio for building energy and daylighting simulation. The Honeybee project aims to make the features of these simulation tools available parametrically and with the flexibility offered by a component-based, visual scripting interface. (Sadeghipour Roudsari M., Pak M., 2013).

Roudsari, Mostapha Sadeghipour, Pak, Michelle, 2013. Ladybug: a parametric environmental plugin for grasshopper to help designers create an environmentally-conscious design. In: Proceedings of the 13th International IBPSA Conference Held in Lyon, France Aug 25–30th. (http://www.ibpsa.org/proceedings/BS2013/p_2499.pdf)

Should we also mention that the engines are already validated?

@ayezioro and @arkdanielnielsen can you also please give us your input? Thanks.

**Comment by Kristoffernegendahl** [comment]
Hey everyone. I have been eavesdropping on some of the updates here lately. And I thought I would comment this thread.
When I started my Ph.D I started reading all kinds of articles trying to figure out how I could classify my own plugins, Geco, Ladybug, Honeybee and many other interesting projects. The thing is, I couldn't find any clear definition anywhere. Like @chriswmackey says, terms like "parametric"  and "parametrically" are tossed around frequently, however this term does not (in my opinion at least) cover performace-feedback from the simulation tools.
I started writing a review-article on the subject. It is primarily focusing on the pros and cons of using plugins like LB/HB. This article is pending at Automation in Construction, and hopefully someone would pick it up and read it one day. In the mean time, I thought the work could benefit here.This is what I ended up calling "this parametric plugin-thing": *integrated dynamic models*

![6-figure-19](https://cloud.githubusercontent.com/assets/3524250/6225770/08f7c91e-b68f-11e4-82fb-910c01da8960.png)
 
I ended up naming on the model and not the tools or combination of tools. This is convenient when trying to compare e.g. IESve, which combines the CAD interface (design tool) with the BPS (building performance simulation) tool in a single package. 
The integrated dynamic model is a special case of distributed models which may work in a mesh/web configuration with many other tools. 

![6-figure-6](https://cloud.githubusercontent.com/assets/3524250/6225923/7cc97328-b690-11e4-9056-a1215a5cd360.png)

Of course I could go on and bore you forever with this. Which in no way is my intention. The way I see HB and LB is plugins that enable dynamic couplings between BPS tools and VPLs (Grasshopper and soon Dynamo I understand!?). This creates all kinds of fantastic possibilities only possible with integrated dynamic models ... or what ever they are called!  

I don't know if this has helped in any way at all. If not just disregard everything I said :)

**Comment by ayezioro** [comment]
Hi,
I've found @Kristoffernegendahl comments very interesting and relevant for the paragraph you want to write.
I think that the "performance" issue is important to be included, probably at the beginning, kind of: 
Ladybug and Honeybee are two open source plugins for Grasshopper3D for Rhino for environmental performance evaluation.
I would also mention that the simulation engines are validated. Yes, i call them simulation engines, even though they can be beyond that but almost no one uses them as interfaces or such.
As a personal note i can say that as much validated they are, they don't guarantee quality results. As i mentioned somewhere else: "Garbage in, garbage out". As much as such tools can open doors and windows when used at "total" control, they can also lead to "catastrophes" when they don't. Here the issue of education and understanding appear as a main column of the simulation and the tools are only a mean for better understanding and better designing.
I rest here ...
-A.

**Comment by mostaphaRoudsari** [comment]
Thank you @Kristoffernegendahl  and @ayezioro- If not anything else I love the Internet for moments like this. You don't know who knows the answer but you wonder someone should be around who knows!

I liked the notion of "integrated dynamic model" very much, and I think distributed model is what I mean when I show the "Welcome to the zoo!" slide (http://goo.gl/Pq3uw7) Now I have a better term to put it forward. I'm looking forward to read your paper when it is published. Let me know when it is available.

Ok. Let me try again! Now we need to reference Kristoffer's paper in ladybug's reference. I think we are making kind of distributed model here in referencing too! ;) I also wonder if readers will understand "integrated dynamic model" in this context. Nevertheless I will give it a try and we can discuss it afterward.

Ladybug and Honeybee are two open source plugins for environmental performance evaluation for Grasshopper3D for Rhino. Ladybug imports standard EnergyPlus Weather files (.EPW) into Grasshopper3D and provides a variety of 3D interactive graphics to support the decision-making process during the initial stages of design. Honeybee connects visual programming environment of Grasshopper3D to validated simulation engines, particularly, EnergyPlus, Radiance, Daysim and OpenStudio for building energy, comfort and daylighting simulation (Sadeghipour Roudsari M., Pak M., 2013). These plugins aim to enable dynamic couplings between flexible component-based visual programming interfaces and validated building performance simulation engines. 

Roudsari, Mostapha Sadeghipour, Pak, Michelle, 2013. Ladybug: a parametric environmental plugin for grasshopper to help designers create an environmentally-conscious design. In: Proceedings of the 13th International IBPSA Conference Held in Lyon, France Aug 25–30th. (http://www.ibpsa.org/proceedings/BS2013/p_2499.pdf)

Such a mixture! I will read it again tomorrow morning! Feel free to comment on the current version.

**Comment by ayezioro** [comment]
Reads fine!
I'll also like to read @Kristoffernegendahl when published.

**Comment by Kristoffernegendahl** [comment]
@mostaphaRoudsari I think it sounds good. And thank you guys at least two people will eventually read it :) I will send you a copy of the article; **Building Performance Simulation in the early design stage: An introduction to Integrated Dynamic Models** as soon as Automation in Construction lets me know when it is finally accepted.

**Comment by ayezioro** [comment]
Not only two. Next semester i'll teach a class on simulations. So i'll give the students the paper as lecture reading ... hoping A&C will accept it by then :-)
-A.

**Comment by Kristoffernegendahl** [comment]
@ayezioro :)

**Comment by mostaphaRoudsari** [comment]
@Kristoffernegendahl I think you should post it on the group once it is published. I'm pretty sure there are so many people out there who love to read it and give you feedback.

**Comment by Kristoffernegendahl** [comment]
@mostaphaRoudsari Yes you are right. I will do that. Now I hope Automation in Construction accepts it soonish.

**Comment by mostaphaRoudsari** [comment]
Thank you everyone. I added a file here as a reference and I'm closing this one.
https://github.com/mostaphaRoudsari/ladybug/blob/master/resources/How_to_Reference_Ladybug_and_Honeybee.md

**Comment by chriswmackey** [comment]
Very sorry for returning to the discussion so late.  The vocabulary of the description sounds great and thank you for the insightful input, @Kristoffernegendahl .  There are just a few things that the description is missing to make it grammatically correct (mostly, it just needs some "the" 's and "a" 's and some way of avoiding three "for" 's in a sentence).  Also, I had always thought of Rhino as the program that needed the "3D" at the end of it (as opposed to Grasshopper).  Let me know if this sounds good:

Ladybug and Honeybee are two open source plugins for Grasshopper and Rhino3D that help explore and evaluate environmental performance. Ladybug imports standard EnergyPlus weather files (.EPW) into Grasshopper and provides a variety of 3D interactive graphics to support the decision-making process during the initial stages of design. Honeybee connects the visual programming environment of Grasshopper to four validated simulation engines - specifically, EnergyPlus, Radiance, Daysim and OpenStudio - which evaluate building energy consumption, comfort, and daylighting (Sadeghipour Roudsari M., Pak M., 2013). These plugins enable a dynamic coupling between the flexible, component-based, visual programming interface of Grasshopper and validated environmental data sets and simulation engines.

Let me know if you would like me to make the change, @mostaphaRoudsari , and sorry again for not responding sooner.
-Chris

**Comment by mostaphaRoudsari** [comment]
Hey @chriswmackey! Thank you for the corrections. I just checked the websites for both Rhino and Grasshopper and technically they're "Rhinoceros" and "Grasshopper" and there is no 3D for neither of them. Should we change "Rhino3D" to "Rhinoceros"?

**Comment by chriswmackey** [comment]
@mostaphaRoudsari ,
The suggested correction sounds good and I have just changed it in the document.  I think that a lot of people put the "3D" in there to clarify what Rhino is to people who aren't familiar with it (similar to the way that Arpan was using the word "CAD" to describe a lot of projects that were Rhino-related).  I have come to realize that, if people aren't familiar with Rhino, then they are probably just not going to understand what Ladybug and Honeybee are so I agree with the edit.  I have taken it out in a recent commit of the text.
-Chris

**Comment by mostaphaRoudsari** [comment]
Thanks @chriswmackey!

**Comment by Kristoffernegendahl** [comment]
@mostaphaRoudsari @ayezioro and **@everyoneelse**
Here is a notification as promised, my jounal paper has finaly been published online: Free to read next 10 days: http://authors.elsevier.com/a/1QmIL3IhXMSz0n

**Comment by ayezioro** [comment]
Thanks Kristoffer.
Downloaded already.
-A.

**Comment by mostaphaRoudsari** [comment]
@Kristoffernegendahl I finally read your paper last night. Great overview overall. I liked the diagrams very much and the distinctions that you made between different types of simulation tools is very clear and helpful. I usually find myself having a hard time to describe how Honeybee is structured differently versus some other simulation tools to non-experts. Now I will be using your diagrams to make it clear. Thanks.

**Comment by Kristoffernegendahl** [comment]
@mostaphaRoudsari I am glad to be of any use. And you are very welcome to use the articles as you please.
